selfservice/flow/login/handler.go

```diff
@@ -765,22 +778,86 @@ func fetchUserProfileFromToken(p *LifepalOauthLoginPayload) (*GoogleResponseProf
 	return structResp, nil
 }
 
+var (
+	fireConnOnce sync.Once
+	firebaseConn = new(auth.Client)
+)
+
+func GetFirebaseConnection() (*auth.Client, error) {
+	opt := option.WithCredentialsFile(getFirebaseCredential())
+	app, err := firebase.NewApp(context.Background(), nil, opt)
+	if err != nil {
+		return nil, errors.WithStack(errors.New(fmt.Sprintf("error initializing app: %v\n", err)))
+	}
+	// Access Auth service from default app
+	client, err := app.Auth(context.Background())
+	if err != nil {
+		return nil, errors.WithStack(errors.New(fmt.Sprintf("error getting Auth client: %v\n", err)))
+	}
+	return client, nil
+}
+
+func initConnection() *auth.Client{
+	fireConnOnce.Do(func() {
+		firebaseConn, _ = GetFirebaseConnection()
+	})
+	return firebaseConn
+}
+
+func FetchUserFromFirebase(p *LifepalOauthLoginPayload) (*auth.UserRecord, error) {
+	client := initConnection()
+	u, err := client.GetUser(context.Background(), p.Identifier)
+	if err != nil {
+		return nil, errors.WithStack(errors.New(fmt.Sprintf("error getting user %s: %v\n", p.Identifier, err)))
+	}
+
+	// check if phone number is valid
+	if u.PhoneNumber[:3] != "+62" {
```

Comment:
i think currently on userauth we strip `+62` or `0` on the beginning on phone number. but we could update this when we integrate with gatekeeper